internal/services/producer.go

```diff
@@ -109,6 +124,10 @@ func (ps *producerService) Send(partitionsAssignments map[int32][]int32) {
 			"clientid":  ps.canaryConfig.ClientID,
 			"partition": strconv.Itoa(i),
 		}
+		// initialize all error-related metrics with starting value of 0
+		if !ps.initialized {
+			recordsProducedFailed.With(labels).Add(0)
+		}
```

Comment:
can we apply the same argument I made before for the connection check here?
I mean getting rid of `initialized` and `previousNumPartitions` then, if there is an error on sending you have `recordsProducedFailed.With(labels).Inc()` called. In the `else` branch where the latency is computed, you can just add `recordsProducedFailed.With(labels).Add(0)` which initilized the metric for new partitions or just doing nothing for existing ones.
I guess it should work. Wdyt?